sdk/python/feast/registry.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+from abc import ABC, abstractmethod
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryFile
+from typing import Callable, List
+from urllib.parse import urlparse
+
+from google.auth.exceptions import DefaultCredentialsError
+
+from feast.core.Registry_pb2 import Registry as RegistryProto
+from feast.entity import Entity
+from feast.feature_table import FeatureTable
+
+REGISTRY_SCHEMA_VERSION = "1"
+
+
+class Registry:
```

Comment:
Would you mind please adding [Python docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to classes and methods? You can use your judgement on how much/little detail you want to provide, since some of the methods are documented elsewhere.